src/main/webapp/script.js

```diff
@@ -33,3 +33,733 @@ function createCommentElement(comment) {
   commentElement.appendChild(userComment);
   return commentElement;
 }
+
+/** Fetches results from the server and adds them to the DOM. */
+function getResults(param) {
+  var userQuery = getURLParamVal(param);
+  // Search string is in all caps, so the userQuery should also be in all caps for querying purposes.
+  fetch('/results?user-query=' + userQuery.toUpperCase()).then(response => response.json()).then((results) => {
+    const resultListElement = document.getElementById('result-list');
+    results.forEach((result) => {
+      console.log("Result found!");
+      resultListElement.appendChild(createResultElement(result));
+    })
+  });
+}
+
+/** Gets the ID of a YouTube video from its URL.
+    Will work only for a URL of a certain format
+    (i.e. youtube.com/watch?v=). */
+function getIdFromUrl(url) {
+  return url.substring(url.lastIndexOf('=') + 1);
+}
+
+/** Gets the ID of the YouTube video that the user inputs. */
+function getId() {
+  const liveStreamLink = document.getElementById('live-stream-link').value;
+  return getIdFromUrl(liveStreamLink);
+}
+
+/** Gets the ID of the YouTube video that the user inputs. */
+function storeLiveStreamInfo(schedStartTime, schedEndTime) {
+  const recipeSelection = document.getElementById('recipe-selection');
+  const recipeKey = recipeSelection.options[recipeSelection.selectedIndex].text;
+  const liveStreamLink = document.getElementById('live-stream-link').value;
+  fetch('/new-live-stream?recipe-key=' + recipeKey + '&live-stream-link=' + liveStreamLink + '&sched-start-time=' + schedStartTime + '&sched-end-time=' + schedEndTime);
+}
+
+/** Videos: List JSON Response Retrieval */
+// https://apis.google.com/js/api.js
+var gapi = window.gapi = window.gapi || {};
```

Comment:
is this all API stuff? If so I'll just assume it works

Reply:
yes LOL